dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricRegistryProducerAdapter.java

```diff
@@ -0,0 +1,85 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.smallrye.metrics.cdi.adapters;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.enterprise.inject.Default;
+import jakarta.enterprise.inject.Produces;
+import jakarta.enterprise.inject.spi.InjectionPoint;
+
+import org.eclipse.microprofile.metrics.MetricRegistry;
+
+import com.ibm.websphere.ras.Tr;
+import com.ibm.websphere.ras.TraceComponent;
+
+import io.openliberty.microprofile.metrics50.helper.Util;
+
+/**
+ *
+ */
+@ApplicationScoped
+public class MetricRegistryProducerAdapter {
+    private static final TraceComponent tc = Tr.register(MetricRegistryProducerAdapter.class);
+    static Class<?> srMetricsProducerClass;
+    static Object srMetricsProducerObj;
+
+    static {
+        srMetricsProducerClass = Util.SR_METRIC_REGISTRY_PRODUCER_CLASS;
+
+        if (srMetricsProducerClass == null) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "The SmallRye MetricRegistryProduer class was not resolved.");
+                // Use an Error message like the following?!?!
```

Comment:
Is this still needed?

Reply:
The comment was left for discussion. Whether we use a debug statement if the class isn't loaded or throw a error message to be translated.